public/app/plugins/panel/timeseries/plugins/OutsideRangePlugin.tsx

```diff
@@ -25,37 +26,41 @@ export const OutsideRangePlugin = ({ config, onChangeTimeRange }: ThresholdContr
     });
   }, [config]);
 
+  // If we don't have enough time values or if we can't
+  // change the time range then we don't display zoom to data
   if (timevalues.length < 2 || !onChangeTimeRange) {
     return null;
   }
 
+  // If we don't have an appropriate time range then we don't know if data is
+  // being displayed
   if (!timeRange || !timeRange.time || !timeRange.min || !timeRange.max!) {
     return null;
   }
 
   // Time values are always sorted for uPlot to work
-  let i = 0,
-    j = timevalues.length - 1;
+  let i = 0;
+  let j = timevalues.length - 1;
 
-  while (i <= j && timevalues[i] == null) {
-    i++;
-  }
+  // Increment i until we get the last non-null value
+  for (; i < timevalues.length && timevalues[i] == null; i++) {}
```

Comment:
i can end up at out of bounds of array after this loop? (equal to `timevalues.length`)?

Reply:
<img width="269" alt="image" src="https://github.com/grafana/grafana/assets/43234/acfb279d-e7f7-4e3e-be3e-3ac3ce2fdfa9">

Reply:
Arrrr, haha you go me 💀

Reply:
Updated these conditions, if you could have another look it would be appreciated 🙏